SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowCapabilitySpec.m

```diff
@@ -5,60 +5,52 @@
-__block SDLSoftButtonCapabilities *testSoftButtonsCapabilities = nil;
-__block SDLImageType testImageType = nil;
-__block NSString *testTextName = nil;
-__block NSString *testImageName = nil;
-__block NSString *testTemplateAvailable = nil;
-__block SDLMenuLayout testMenuLayout = SDLMenuLayoutTiles;
-__block SDLDynamicUpdateCapabilities *testDynamicUpdates = nil;
-
-describe(@"Getter/Setter Tests", ^ {
-    beforeEach(^{
-        testImageType = SDLImageTypeDynamic;
-        testTextName = @"test Text field";
-        testImageName = @"test Image field";
-        
-        testTextField = [[SDLTextField alloc] init];
-        testTextField.name = SDLTextFieldNameTertiaryText;
-        testImageField = [[SDLImageField alloc] init];
-        testImageField.name = testImageName;
-        
-        testButtonCapabilities = [[SDLButtonCapabilities alloc] init];
-        testButtonCapabilities.name = SDLButtonNameOk;
-        testButtonCapabilities.shortPressAvailable = @YES;
-        testButtonCapabilities.longPressAvailable = @YES;
-        testButtonCapabilities.upDownAvailable = @YES;
-        
-        testSoftButtonsCapabilities = [[SDLSoftButtonCapabilities alloc] init];
-        testSoftButtonsCapabilities.imageSupported = @YES;
+NSString *testTemplateAvailable = @"myTemplate";
+SDLTextField *testTextField = [[SDLTextField alloc] init];
+NSArray<SDLTextField *> *textFields = @[testTextField];
+NSString *textFieldName = @"t.f.name";
+testTextField.name = textFieldName;
+SDLImageField *testImageField = [[SDLImageField alloc] init];
+NSArray<SDLImageField *> *imageFields = @[testImageField];
+NSString *testImageName = @"test Image field";
+testImageField.name = testImageName;
+SDLButtonCapabilities *testButtonCapabilities = [[SDLButtonCapabilities alloc] init];
+NSArray<SDLButtonCapabilities *> *buttonCapabilities = @[testButtonCapabilities];
+testButtonCapabilities.name = SDLButtonNameOk;
+testButtonCapabilities.shortPressAvailable = @YES;
+testButtonCapabilities.longPressAvailable = @YES;
+testButtonCapabilities.upDownAvailable = @YES;
+SDLSoftButtonCapabilities *testSoftButtonsCapabilities = [[SDLSoftButtonCapabilities alloc] init];
+testSoftButtonsCapabilities.imageSupported = @YES;
+SDLImageType testImageType = SDLImageTypeDynamic;
+SDLMenuLayout testMenuLayout = SDLMenuLayoutTiles;
+SDLDynamicUpdateCapabilities *testDynamicUpdates = [[SDLDynamicUpdateCapabilities alloc] initWithSupportedDynamicImageFieldNames:@[SDLImageFieldNameSubMenuIcon] supportsDynamicSubMenus:@YES];
+SDLKeyboardCapabilities *keyboardCapabilities = [[SDLKeyboardCapabilities alloc] init];
+id windowID = @444;
+id numCustomPresetsAvailable = @10;
+NSArray<SDLImageType> *imageTypeSupported = @[testImageType];
+NSArray<NSString *> *templatesAvailable = @[testTemplateAvailable];
+NSArray<SDLSoftButtonCapabilities *> *softButtonCapabilities = @[testSoftButtonsCapabilities];
+NSArray<SDLMenuLayout> *menuLayoutsAvailable = @[testMenuLayout];
```

Comment:
Is it more clear to stick to the pattern of initializing these inside the describe block?

Reply:
The describe block can be called multiple times when I expect all variables to be init once only and never change (more like const or final expression)